### PR TITLE
storage: Rewrite FSYS and NFS tables in storage overview to use ListingTable (PF4)

### PR DESCRIPTION
**Pull request by KKoukiou**

### CODE REVIEW

**Comment by KKoukiou** [comment]
With this PR they tables look like this:
Screen Shot 2019-11-06 at 14 27 58

File: pkg/storaged/fsys-panel.jsx [hunk 1]
```diff
@@ -62,53 +65,50 @@ export class FilesystemsPanel extends React.Component {
...
-                    </td>
-                </tr>
-            );
+                    }
+                ]
+            };
         }
 
         var mounts = Object.keys(client.blocks).filter(is_mount)
                 .sort(cmp_mount)
```
Comment (mvollmer) [inline pkg/storaged/fsys-panel.jsx]: Do we still need to sort the rows explicitly?  Or does the table sort them anyway?

Comment (KKoukiou) [reply pkg/storaged/fsys-panel.jsx]: AH good catch _ I can make the table have some initial sorted state without this.

File: pkg/lib/cockpit-components-table.jsx [hunk 2]
```diff
@@ -175,11 +183,9 @@ export class ListingTable extends React.Component {
     }
 }
 ListingTable.defaultProps = {
-    caption: '',
```
Comment (mvollmer) [inline pkg/lib/cockpit-components-table.jsx]: Why did you change the default props?

File: pkg/lib/cockpit-components-table.less [hunk 3]
```diff
@@ -6,6 +6,7 @@ header {
     }
 }
 
+
```
Comment (mvollmer) [inline pkg/lib/cockpit-components-table.less]: Noise?

File: pkg/storaged/fsys-panel.jsx (hunk 1 above)
Comment (KKoukiou) [reply pkg/storaged/fsys-panel.jsx]: Fixed.

File: pkg/lib/cockpit-components-table.jsx (hunk 2 above)
Comment (KKoukiou) [reply pkg/lib/cockpit-components-table.jsx]: Moved to previous state.

File: pkg/lib/cockpit-components-table.less (hunk 3 above)
Comment (KKoukiou) [reply pkg/lib/cockpit-components-table.less]: Fixed.

**Comment by andreasn** [comment]
The headers look very big compared to the rows now, but looking at the PF4 design guidelines that is the correct behavior.
Looking at the [PF4 tables](https://www.patternfly.org/v4/documentation/react/components/table), it looks like it might work better inside a card. Made a quick mockup of that. What do you think?
storage-pf4

**Comment by andreasn** [comment]
Also curious to hear what @mvollmer and @garrett think about that style on the storage page.

**Comment by andreasn** [comment]
So yeah, we can restyle it we feel it would look better, but I'm OK with it as it is now too.

**Comment by lachlan-00** [comment]
Thanks for this one!
